server/controllers/commentController.js

```diff
@@ -58,7 +58,7 @@ const getComment = async (req, res) => {
 // @returns {object} 500 - Returns an object that returns a message "Internal Server Error"
 const deleteComment = async (req, res) => {
 
-    if ((req.query.postUserId !== req.query.userId) || (req.query.postUserId !== req.user._id.toString()))
+    if ((req.query.postUserId !== req.user._id.toString()) && (req.query.userId !== req.user._id.toString()))
```

Comment:
usually these conditions gets too complex and unreadable. it is better to extract to a variable for others to understand what is happening here.